topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py

```diff
@@ -109,6 +106,31 @@ def read_csv(self, metadata_file: str = "") -> None:
 
         self.is_init = True
 
+    def add_centroid(self, item: Item, asset_metadata: Dict[str, str]):
+        centroid = {
+            "lat": string_to_number(asset_metadata.get("photocentre_lat", None)),
+            "lon": string_to_number(asset_metadata.get("photocentre_lon", None)),
+        }
+        if not all(centroid.values()) or "" in centroid.values():
+            item.add_warning(
+                msg="Skipped Record",
+                cause=self.name,
+                e=Exception(f"stac field 'proj:centroid' has None values: {centroid}"),
+            )
+        else:
+            item.properties.update({"proj:centroid": centroid})
+            item.add_extension(stac.StacExtensions.projection.value)
+
+    def add_mission(self, item: Item, asset_metadata: Dict[str, str]):
+        survey = asset_metadata.get("survey", None)
+        if not survey or survey == "0" or survey == "":
+            alternative_survey = asset_metadata.get("alternative_survey", None)
+            if alternative_survey and alternative_survey != "":
```

Comment:
In the other case `mission` will never be added. As it is a required field, do we want a warning for that?

Reply:
we could put a warning here, I had left it for the metadata_validator_stac to pick it up?

Reply:
I agree with that approach too. However we should clarify this with @dwsilk and apply this (leaving the jsonschema validation to raise the error) to every required field.